agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -133,8 +135,55 @@ static bool get_media_type(const std::string &interface_name,
             result = true;
         }
     } else if (ieee1905_1::eMediaTypeGroup::IEEE_802_11 == media_type_group) {
-        // TODO: Not supported yet
-        LOG(ERROR) << "IEEE_802_11 media is not supported yet";
+        /**
+         * Value and Description         Frequency  MaxBandwith  Comments
+         * 0 IEEE 802.11b (2.4 GHz)      2.4 GHz    22 MHz       Not supported
+         * 1 IEEE 802.11g (2.4 GHz)      2.4 GHz    20 MHz
+         * 2 IEEE 802.11a (5 GHz)        5 GHz      20 MHz
+         * 3 IEEE 802.11n (2.4 GHz)      2.4 GHz    40 MHz
+         * 4 IEEE 802.11n (5 GHz)        5 GHz      40 MHz
+         * 5 IEEE 802.11ac (5 GHz)       5 GHz      80 MHz
+         * 6 IEEE 802.11ad (60 GHz)      60 GHz     160 MHz      Not supported
+         * 7 IEEE 802.11af (whitespace)                          Not supported
+         * 8 IEEE 802.11ax (2.4 GHz)     2.4 GHz    160 MHz      Not included in Table 6-12—Media type (intfType), WiFi6 is specified to use 0x0108 in R2
+         * 8 IEEE 802.11ax (5 GHz)       5 GHz      160 MHz      Not included in Table 6-12—Media type (intfType), WiFi6 is specified to use 0x0108 in R2
+         */
+
+        std::unique_ptr<bwl::nl80211_client> client =
+            bwl::nl80211_client_factory::create_instance();
+        bwl::nl80211_client::radio_info radio_info;
+
+        if (client->get_radio_info(interface_name, radio_info)) {
+            if (!radio_info.bands.empty()) {
+                bwl::nl80211_client::band_info band_info = radio_info.bands.at(0);
+
+                beerocks::eFreqType frequency_band     = band_info.get_frequency_band();
+                beerocks::eWiFiBandwidth max_bandwidth = band_info.get_max_bandwidth();
+
+                if ((beerocks::eFreqType::FREQ_24G == frequency_band) &&
```

Comment:
This block is really hard to read.
Maybe you could first group by frequency, then by bandwidth.
You could also maybe use a switch for the bandwidth
(nitpick)

Reply:
Done